### PR TITLE
[FIX] website_sale_collect,*: fix flow when warehouse is set on website

### PR DESCRIPTION
**Pull request by vchu-odoo**
When warehouse is set on website then website_sale_stock adds logic to handle out of stock products, namely preventing to sell them or adding to the cart. However, when click and collect is activated user can choose a warehouse where the product is available that is not necessarily the one set on the website. For these cases user should be capable to add products in the cart. The validation of the availability is checked later in the /checkout and /payment.

### CODE REVIEW

**Review by ltinel — CHANGES_REQUESTED** [review]
Thanks for your work! I'm already posting a first round of reviews but still need to test locally :) I also think it might be worth adding a few automated tests (maybe in a follow-up PR if this one is urgent)

File: addons/delivery/static/src/js/location_selector/location_selector_dialog/location_selector_dialog.js
Comment (vchu-odoo) [reply addons/delivery/static/src/js/location_selector/location_selector_dialog/location_selector_dialog.js:156]: yes, but that is a location selector logic. Need to ask vcr :)

File: addons/website_sale_collect/models/sale_order.py
Comment (vchu-odoo) [reply addons/website_sale_collect/models/sale_order.py:14]: We have chosen 'delivery method' as the most appropriate name and we try to promote it wherever is possible. So on the contrary :)

File: addons/website_sale_collect/models/sale_order.py
Comment (vchu-odoo) [reply addons/website_sale_collect/models/sale_order.py:13]: Indeed, good catch!

File: addons/website_sale_collect/models/sale_order.py
Comment (vchu-odoo) [reply addons/website_sale_collect/models/sale_order.py:86]: the check should be only for storable, I can use the full condition for consistency.

File: addons/delivery/static/src/js/location_selector/location_selector_dialog/location_selector_dialog.js
Comment (ltinel) [reply addons/delivery/static/src/js/location_selector/location_selector_dialog/location_selector_dialog.js:156]: I'll ask him. Small nit: missing `;` at the EOL.

**Review by ltinel — CHANGES_REQUESTED** [review]
Almost done :) Could you also add an override for the combo configurator?

**Review by ltinel — APPROVED** [review]
Looks good! Could you also rename your PR/commit message to include all updated modules? Thanks! :)

File: addons/website_sale_stock/views/res_config_settings_views.xml
Comment (chevalierv) [inline addons/website_sale_stock/views/res_config_settings_views.xml:17]: Don't forget to add the placeholder back in https://github.com/odoo/odoo/pull/181383

File: addons/website_sale_collect/models/sale_order.py
Comment (chevalierv) [inline addons/website_sale_collect/models/sale_order.py:54]: Why don't use `self.warehouse_id` now that you set it when selecting the location?
```suggestion
            and not self._is_in_stock(self.warehouse_id)
```

File: addons/website_sale_collect/models/sale_order.py
Comment (chevalierv) [inline addons/website_sale_collect/models/sale_order.py:99]: `_verify_updated_quantity` calls `_get_cart_and_free_qty` in `website_sale_stock` which use the warehouse of the SO. Therefore I don't think this override is necessary.

File: addons/website_sale_collect/static/src/js/variant_mixin.js
Comment (chevalierv) [inline addons/website_sale_collect/static/src/js/variant_mixin.js:14]: This should not be needed. In the current codebase, there's a discrepancy between `_get_product_available_qty` and `_get_cart_and_free_qty`. They may not use the same warehouse to compute the quantities. 

I don't see how to ensure using the same warehouse as we don't have access to the SO in `_get_additionnal_combination_info` so let's keep it like that as of now.

**Review by chevalierv — CHANGES_REQUESTED** [review]
Hello @vchu-odoo 

I've got a few comments/questions otherwise LGTM :+1: 

While reviewing, I became aware that you sent the quantity to the location selector even when the show quantity was hidden. IMO it's not a good practice because the information is available in the network tab and the console. I believe it would be better (maybe not in this PR if it takes more than 10 minutes to add it) to not send the quantity at all, and guess that if there is no quantity, you shouldn't show anything.

Have a nice day!

File: addons/website_sale_collect/controllers/main.py
Comment (vchu-odoo) [reply addons/website_sale_collect/controllers/main.py:34]: The template itself still shows unavailable products from product lines in the end. So if you want I can rename the template but it does not change much

File: addons/website_sale_stock/views/res_config_settings_views.xml
Comment (vchu-odoo) [reply addons/website_sale_stock/views/res_config_settings_views.xml:17]: Will add after merging and rebasing on this PR

File: addons/website_sale_collect/models/sale_order.py
Comment (vchu-odoo) [reply addons/website_sale_collect/models/sale_order.py:99]: but when the customer has not selected a pickup point yet? it will use the one set on the website and it will not allow not add it.

File: addons/website_sale_collect/static/src/js/variant_mixin.js
Comment (vchu-odoo) [reply addons/website_sale_collect/static/src/js/variant_mixin.js:14]: it is needed when pickup location is not selected yet

File: addons/website_sale/static/src/js/quantity_buttons/quantity_buttons.xml
Comment (vchu-odoo) [reply addons/website_sale/static/src/js/quantity_buttons/quantity_buttons.xml:6]: small fix, thought could be in the same commit. Will separate :)

File: addons/website_sale_collect/models/sale_order.py
Comment (chevalierv) [reply addons/website_sale_collect/models/sale_order.py:101]: Good point. I didn't think about that use case... I believe it's weird as it is right now but I think it would be worse to prevent adding and item to the cart so we'll improve that later.

File: addons/website_sale_collect/views/delivery_form_templates.xml
Comment (chevalierv) [inline addons/website_sale_collect/views/delivery_form_templates.xml:13]: :eyes: 
As per my IDE, the `<a>` and the `<t>` tags below are well positioned, maybe you want to fix the `<div>` just above?

```suggestion
                    <a t-att-href="order_line.product_id.website_url">
```
image

**Review by chevalierv — APPROVED** [review]
Hello @vchu-odoo 

LGTM :+1: You can ignore the small nitpicking below

@robodoo delegate+

Have a nice day!

File: addons/website_sale_collect/static/src/js/website_sale_configurators.js
Comment (chevalierv) [reply addons/website_sale_collect/static/src/js/website_sale_configurators.js:13]: Discard my previous comment, It's currently not possible to do it in 18.0 and I don't want to hold back all the nice features in this PR just because this one doesn't work. We'll improve that later :)

**Comment by vchu-odoo** [comment]
@robodoo rebase-ff

**Comment by vchu-odoo** [comment]
@robodoo r+

**Pull request closed without merging**
